### PR TITLE
[seekhandler] fix wrong seek step processing if no delay is configured

### PR DESCRIPTION
**Pull request by xhaggi**
This will refactors the seek step processing and reduce the roundtrips we need to process a seek with no delay or the built-in seek by seconds. It also fix the wrong seek size calculation if you hold the seek button down for a period of time while you have no seek delay configured.

@FernetMenta mind taking a look. I hope it's the last time :smile:

### CODE REVIEW

File: xbmc/utils/SeekHandler.cpp [hunk 1]
```diff
@@ -90,45 +90,42 @@ void CSeekHandler::Reset()
...
   CSingleLock lock(m_critSection);
 
   // not yet seeking
   if (!m_requireSeek)
   {
-    // tell info manager that we have started a seek operation
+    // use only the first step forward/backward for a seek without a delay
+    if (!analogSeek && m_seekDelays.at(type) == 0)
+    {
+      SeekSeconds(GetSeekStepSize(type, forward ? 1 : -1));
```
Comment (FernetMenta) [inline xbmc/utils/SeekHandler.cpp]: just a brief look during a break here. I don't think we should hold the lock here

Comment (xhaggi) [inline xbmc/utils/SeekHandler.cpp]: leave it, done in a fixup commit

**Comment by FernetMenta** [comment]
analog seek should not use SeekTimeRelative but SeekTime

**Comment by xhaggi** [comment]
let me know why do you think so

**Comment by FernetMenta** [comment]
> let me know why do you think so

analog seek is when you i.e. click with the mouse into the progress bar. the mouse position is translated  into an absolute time. is this relative to something?

**Comment by xhaggi** [comment]
i thought analog seek is used by controllers like xbox or PlayStation. @da-anda correct me if I'm wrong.

**Comment by FernetMenta** [comment]
wait, I might been wrong here.

**Comment by FernetMenta** [comment]
@xhaggi sorry, I mixed this up, mouse wheel and the like is analog. code is good as you have it

**Comment by xhaggi** [comment]
jenkins build and merge

**Comment by xhaggi** [comment]
@FernetMenta np ;)

**Pull request merged**